server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/get/TransportGetSnapshotsAction.java

```diff
@@ -135,16 +140,23 @@ private void getMultipleReposSnapshotInfo(
     ) {
         // short-circuit if there are no repos, because we can not create GroupedActionListener of size 0
         if (repos.isEmpty()) {
-            listener.onResponse(new GetSnapshotsResponse(Collections.emptyList()));
+            listener.onResponse(new GetSnapshotsResponse(Collections.emptyList(), Collections.emptyMap()));
             return;
         }
-        final GroupedActionListener<GetSnapshotsResponse.Response> groupedActionListener = new GroupedActionListener<>(
-            listener.map(responses -> {
+        final GroupedActionListener<Tuple<Tuple<String, ElasticsearchException>, List<SnapshotInfo>>> groupedActionListener =
+            new GroupedActionListener<>(listener.map(responses -> {
                 assert repos.size() == responses.size();
-                return new GetSnapshotsResponse(responses);
-            }),
-            repos.size()
-        );
+                final List<SnapshotInfo> allSnapshots = responses.stream()
+                    .map(Tuple::v2)
+                    .filter(Objects::nonNull)
+                    .flatMap(Collection::stream)
+                    .collect(Collectors.toUnmodifiableList());
+                final Map<String, ElasticsearchException> failures = responses.stream()
+                    .map(Tuple::v1)
+                    .filter(Objects::nonNull)
+                    .collect(Collectors.toMap(Tuple::v1, Tuple::v2));
+                return new GetSnapshotsResponse(sortSnapshots(allSnapshots, sortBy, after, size, order), failures);
```

Comment:
There's obviously more efficient ways to implement this than just paginating each repo and then merging/sorting as done here. In the interest of time and given that this is just a temporary solution I figured this was good enough though.